src/plugins/interactive_setup/server/kibana_config/kibana_config_service.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { constants } from 'fs';
+import fs from 'fs/promises';
+import yaml from 'js-yaml';
+import path from 'path';
+
+import type { Logger } from 'src/core/server';
+
+import { getDetailedErrorMessage } from '../errors';
+
+export interface WriteConfigParameters {
+  host: string;
+  ca: string;
+  username: string;
+  password: string;
+}
+
+export class KibanaConfigService {
+  constructor(private readonly configPath: string, private readonly logger: Logger) {}
+
+  /**
+   * Checks if we can write to the Kibana configuration file and configuration directory.
+   */
+  public async isConfigWritable() {
+    try {
+      // We perform two separate checks here:
+      // 1. If we can write to config directory to add a new CA certificate file and potentially Kibana configuration
+      // file if it doesn't exist for some reason.
+      // 2. If we can write to the Kibana configuration file if it exists.
+      const canWriteToConfigDirectory = fs.access(path.dirname(this.configPath), constants.W_OK);
+      await Promise.all([
+        canWriteToConfigDirectory,
+        fs.access(this.configPath, constants.F_OK).then(
+          () => fs.access(this.configPath, constants.W_OK),
+          () => canWriteToConfigDirectory
+        ),
+      ]);
+      return true;
+    } catch {
+      return false;
+    }
+  }
+
+  /**
+   * Writes Elasticsearch configuration to the disk.
+   * @param params
+   */
+  public async writeConfig(params: WriteConfigParameters) {
+    const caPath = path.join(path.dirname(this.configPath), `ca_${Date.now()}.crt`);
```

Comment:
__note:__ I decided to skip the check if we can write to that specific cert with (with timestamp in the name) since it's not really actionable, the next enroll attempt will rely on another timestamp.